octobot_evaluators/evaluator/strategy_evaluator.py

```diff
@@ -57,6 +128,201 @@ async def matrix_callback(self,
+                                                  tentacle_name=tentacle_name,
+                                                  tentacle_type=evaluator_type,
+                                                  exchange_name=exchange_name,
+                                                  cryptocurrency=cryptocurrency,
+                                                  symbol=symbol,
+                                                  time_frame=time_frame)
+                                             )
+        self._set_last_evaluation_time(exchange_name, evaluator_type, cryptocurrency, symbol, time_frame, update_time)
+
+    def _already_sent_this_technical_evaluation(self, matrix_id, evaluator, evaluator_type, exchange_name,
+                                                cryptocurrency, symbol, time_frame):
+        try:
+            update_time = get_tentacle_eval_time(matrix_id,
+                                                 get_matrix_default_value_path(
+                                                      tentacle_name=evaluator,
+                                                      tentacle_type=evaluator_type,
+                                                      exchange_name=exchange_name,
+                                                      cryptocurrency=cryptocurrency,
+                                                      symbol=symbol,
+                                                      time_frame=time_frame)
+                                                 )
+            return self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency][symbol][time_frame] \
+                == update_time
+        except KeyError:
+            return False
+
+    def _set_last_evaluation_time(self, exchange_name, evaluator_type, cryptocurrency, symbol, time_frame, value):
+        try:
+            self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency][symbol][time_frame] = value
+        except KeyError:
+            if exchange_name not in self.evaluations_last_updates:
+                self.evaluations_last_updates[exchange_name] = {}
+            if evaluator_type not in self.evaluations_last_updates[exchange_name]:
+                self.evaluations_last_updates[exchange_name][evaluator_type] = {}
+            if cryptocurrency not in self.evaluations_last_updates[exchange_name][evaluator_type]:
+                self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency] = {}
+            if symbol not in self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency]:
+                self.evaluations_last_updates[exchange_name][evaluator_type][cryptocurrency][symbol] = {
+                    time_frame: value
+                }
+
+    def _get_exchange_current_time(self, exchange_name, matrix_id):
+        try:
+            from octobot_trading.api.exchange import get_exchange_current_time, \
```

Comment:
I don't think it's a good idea to make an import during each evaluation. We should check python behavior about imports but I think it's not a good practice.

Reply:
I made some search and it seems to be correctly managed by python, so I think there is no issue here.

Reply:
Agree I'll figure out something

Reply:
oh, perfect then